### ISSUE TITLE
Deleting an item and keyExtractor error

### ISSUE THREAD

**Issue by kashmiry** [computerjazz/react-native-draggable-flatlist#105]
I found this issue when I implemented the functionality to delete an item from my list by removing it from state.data, which work fine. but after that when I sort an item bellow the one I removed, I get an error on the key extractor not being found.

I found out that the **onDragEnd** is returning an undefined for the item I removed. which causes an issue in rerendering.
What I I did is to filter the response and remove the undefined:
`onDragEnd={({ data }) => this.setState({ sortableData: data.filter(function(t){return void 0!==t}) })}`
The issue remain as the **onDragEnd** will still return the old order before deleting the item. the user will have to try again and sort for it to work.

Not sure how to approach this correctly. the onDragEnd need to regenerate the data after the state changes.

This is really a great project, and I am so happy I found it after trying different ones. Thank you so much for everyone who contributed to it 🙏🏼

**Comment by computerjazz** [comment]
Which version are you using? I did recently add some fixes for deletion -- try installing latest (2.0.10 as of now) if you're not already on it.

**Comment by kashmiry** [comment]
@computerjazz Yes I am using the latest (2.0.10). I will report if I find more details about this.

**Comment by mikebianco85** [comment]
HI @computerjazz thanks for the great work! I encountered a similar issue. I used splice to delete an element, and get an error only in certain "patterns". If I delete the last item I don't get an error. if I delete the middle out of three and then drag, the element shoots up (it is not under the finger) and then it gives the error. I was thinking maybe it could be related? or am I maybe doing something wrong? Thanks for your help M
![IMG_5BCDAEA04B3E-1](https://user-images.githubusercontent.com/57252899/75397495-e7a48100-58ee-11ea-94e6-ee565df2abd3.jpeg)
